### PR TITLE
Canonical path

### PR DESCRIPTION
**Pull request by sbrunner**
## Description

Make path absolute to canonicalize them
Display the messages only when it's wanted

## Motivation and Context

Currently, the pylint output sorts the messages to group them by filename, but it's not working because some files are absolute and some other relative.

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
Not sure that removing the make_path function is a good thing, we have a lot of duplication because of it and it's going to be easy to not get the proper relative path if we don't call it like it's expected to be called.

**Comment by sbrunner** [comment]
You prefer what's done in the Test commit?

File: prospector/formatters/pylint.py [hunk 1]
```diff
@@ -32,15 +31,20 @@ def render(self, summary: bool = True, messages: bool = True, profile: bool = Fa
...
                     "function": message.location.function,
                     "message": message.message.strip(),
                 }
             )
+        return output
 
+    def render(self, summary: bool = True, messages: bool = True, profile: bool = False) -> str:
+        output: list[str] = []
+        if messages:
+            output.extend(self.render_messages())
```
Comment (Pierre-Sassoulas) [inline prospector/formatters/pylint.py:47]: This nullify all the argument forcing the default, is  this really  what we want to do ? We can't use formatter interchangeably if we do that. (Breaking the L in SOLID)

Comment (sbrunner) [reply prospector/formatters/pylint.py:47]: > This nullify all the argument forcing the default

I don't understand what you want to say with that?

> We can't use formatter interchangeably if we do that.

To be able to use the formatter interchangeable is my goal, then we have a misunderstanding somewhere!

In fact, it should be possible to move this method to base_summary.py.

Comment (Pierre-Sassoulas) [reply prospector/formatters/pylint.py:47]: I meant previously we could give " summary: bool = True, messages: bool = True, profile: bool = False" to render. I realize now that is wasn't used at all so the behavior doesn't change.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**